### PR TITLE
User guide for custom load balancing policies

### PR DESCRIPTION
**Pull request by temawi**
This is the first new piece of content for the "Guides" section in a while. We are hoping to follow this up with several other guides that highlight the main gRPC features and link to example code.

### CODE REVIEW

**Comment by temawi** [comment]
@ejona86 Roughly what you would expect?

File: content/en/docs/guides/custom-load-balancing.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+more servers.
+
+A gRPC load balancing policy is given a list of server IP addresses by the name
+resolver. The policy is responsible for maintaining connections (subchannels)
+to the servers and picking a connection to use when an RPC is sent.
+
+### Implementing Your Own Policy
+
+By default the `pick_first` policy will be used. This policy actually does no
+load balancing but just uses the first address it gets from the name resolver.
```
Comment (ejona86) [inline content/en/docs/guides/custom-load-balancing.md:22]: It tries each address in order and uses the first that connects.

File: content/en/docs/guides/custom-load-balancing.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+By updating the gRPC service config you can also switch to using `round_robin`
+that rotates through all the addresses it gets. But if the built-in load
```
Comment (ejona86) [inline content/en/docs/guides/custom-load-balancing.md:24]: "that connects to every address it gets and rotates through the connected backends each for each RPC"

Comment (ejona86) [inline content/en/docs/guides/custom-load-balancing.md:24]: Add something about other load balancing policies may exist. Like we have WRR now, and there's least_request in Java. I'm not suggesting to enumerate them all here.

File: content/en/docs/guides/custom-load-balancing.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+balancing policies do not meet your needs (the exact set available vary 
+between languages) you can also implement you own custom policy.
+
+This involves implementing a load balancer interface in the language you are
+using. At a high level, you will have to:
+
+- Register your implementation in the load balancer registry so that it can
+be referred to from the service config
+- Define the JSON configuration object of your implementation. This allows your
```
Comment (ejona86) [inline content/en/docs/guides/custom-load-balancing.md:33]: The definition of this configuration object is sort of virtual. I'd probably emphasize "_parsing_ the JSON configuration object of your implementation."

File: content/en/docs/guides/custom-load-balancing.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+### Service Mesh
+
+If you have a service mesh setup where a central control plane is coordinating
+the configuration of your microservices, you cannot configure your custom load
+balancer directly via the service config. But support is provided to do this
+with the xDS protocol that your control plane uses to communicate with your
+gRPC clients.
+
+For more details, please see gRPC [proposal A52]
```
Comment (ejona86) [inline content/en/docs/guides/custom-load-balancing.md:64]: Fix this link

File: content/en/docs/guides/custom-load-balancing.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+(https://github.com/grpc/proposal/blob/master/A52-xds-custom-lb-policies.md).
+
+### Language Support
+
+| Language | Example | Notes |
+| -------- | ------- | ----- |
+| Java | [link](java-example) | |
```
Comment (ejona86) [inline content/en/docs/guides/custom-load-balancing.md:71]: Use square brackets for the link name `[link][java-example]`. "link" is a very poor link name, so you could do `[java-example][]` which is short-hand for "the text is the same as the link name". You can also have spaces just fine, so can do `[Java example][]` and then at the end have `[Java example]: THE://URL`

File: content/en/docs/guides/custom-load-balancing.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
+---
+title: Custom Load Balancing Policies
+description: >-
+  Explains custom load balancing policies can help optimize load balancing
```
Comment (larry-safran) [inline content/en/docs/guides/custom-load-balancing.md:4]: s/Explains custom/Explains how custom/

File: content/en/docs/guides/custom-load-balancing.md (hunk 2 above)
Comment (temawi) [reply content/en/docs/guides/custom-load-balancing.md]: It was in parenthesis, but I made the point more prominent.

Comment (ejona86) [reply content/en/docs/guides/custom-load-balancing.md]: Yeah, I think the issue is it needed to be before the "But"

File: content/en/docs/guides/custom-load-balancing.md [hunk 7]
```diff
@@ -42,6 +44,16 @@ call path
 The exact steps vary by language, see the language support section for some
 concrete examples in your language.
 
+```mermaid
+flowchart TD
+NR(Name Resolver) -->|Provides addresses &\nLB config| LB(Load Balancer)
+LB --> |Provides a picker| C(Channel)
+C -->|Picks\na subchannel| P(Picker)
```
Comment (ejona86) [inline content/en/docs/guides/custom-load-balancing.md:51]: Would it be better to say the channel "requests a subchannel" from the picker?

**Review by ejona86 — APPROVED** [review]

File: content/en/docs/guides/custom-load-balancing.md (hunk 7 above)
Comment (temawi) [reply content/en/docs/guides/custom-load-balancing.md]: Yes!

**Pull request merged**